.bin/phraseapp/translation-builder.rb

```diff
@@ -71,16 +83,9 @@ def extract_keys_from_tpl_file(file_path)
   # Parses an XML file and returns used keys based on the presence of a specific attribute
   def extract_keys_from_xml_file(file_path)
     file_content = File.read(file_path, :encoding => 'utf-8')
-    doc = Nokogiri::XML(file_content)
-
-    keys = []
-    doc.xpath("//*[@pg_translate='1']").each do |node|
-      if node.key?('id')
-        keys.push(node.attr('id'))
-      end
-    end
-
-    keys
+    keys = file_content.scan(/id="([^"]+)"/).flatten
```

Comment:
This _should_ work most of the time. However, if you have a wonderful & robust XML parser like Nokogiri, it's better to rely on it, instead of using your own regex 🙂

You can keep Nokogiri, and change the **xpath selector** to `'//*'`. This will give you all elements from the XML document. Then you can collect the ids that start with the prefix.

Like so:
```ruby
keys = []

doc = Nokogiri::XML(file_content)
doc.xpath('//*').each do |node|
  if node.key?('id') && node.attr('id').match?(/^#{@locale_prefix}/)
    keys.push(node.attr('id'))
  end
end
```

Reply:
Or, maybe even better, you could try filtering via the xpath function [`starts-with`](https://stackoverflow.com/questions/1556028/how-do-i-do-a-regex-search-in-nokogiri-for-text-that-matches-a-certain-beginning):

```ruby
doc.xpath("//*[starts-with(@id, '#{@locale_prefix}')]")
```

And in this case, you could throw `map` into the mix and make this an elegant one-liner that gives you the `keys` array directly 😎:

```ruby
keys = doc.xpath("//*[starts-with(@id, '#{@locale_prefix}')]").map { |node| node.attr('id') }
```
(not tested, but should work)